components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -530,6 +530,212 @@ func (backend *ES2Backend) GetReport(esIndex string, reportId string,
+		FetchSource(false).
+		Size(1)
+
+	controlTermsAgg := elastic.NewTermsAggregation().Field("profiles.controls.id").
+		Size(100).
+		Order("_key", true)
+
+	controlTermsAgg.SubAggregation("impact",
+		elastic.NewTermsAggregation().Field("profiles.controls.impact").Size(1))
+
+	controlTermsAgg.SubAggregation("title",
+		elastic.NewTermsAggregation().Field("profiles.controls.title").Size(1))
+
+	controlTermsAgg.SubAggregation("skipped",
+		elastic.NewFilterAggregation().Filter(elastic.NewTermQuery("profiles.controls.status", "skipped")))
+
+	controlTermsAgg.SubAggregation("failed",
+		elastic.NewFilterAggregation().Filter(elastic.NewTermQuery("profiles.controls.status", "failed")))
+
+	controlTermsAgg.SubAggregation("passed",
+		elastic.NewFilterAggregation().Filter(elastic.NewTermQuery("profiles.controls.status", "passed")))
+
+	controlTermsAgg.SubAggregation("end_time", elastic.NewReverseNestedAggregation().SubAggregation("most_recent_report",
+		elastic.NewTermsAggregation().Field("end_time").Size(1)))
+
+	profileInfoAgg := elastic.NewReverseNestedAggregation().Path("profiles").SubAggregation("version",
+		elastic.NewTermsAggregation().Field("profiles.name"))
+
+	profileInfoAgg.SubAggregation("sha",
+		elastic.NewTermsAggregation().Field("profiles.sha256"))
+
+	profileInfoAgg.SubAggregation("title",
+		elastic.NewTermsAggregation().Field("profiles.title"))
+
+	profileInfoAgg.SubAggregation("version",
+		elastic.NewTermsAggregation().Field("profiles.version"))
+
+	controlTermsAgg.SubAggregation("profile", profileInfoAgg)
+
+	//todo handle passed in regex too
+	filteredControls := elastic.NewFilterAggregation().Filter(elastic.NewBoolQuery().Must(elastic.NewRegexpQuery("profiles.controls.id", ".*")))
+
+	filteredControls.SubAggregation("control", controlTermsAgg)
+
+	controlsAgg := elastic.NewNestedAggregation().Path("profiles.controls")
+	controlsAgg.SubAggregation("filtered_controls", filteredControls)
+
+	searchSource.Aggregation("controls", controlsAgg)
+
+	source, err := searchSource.Source()
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("%s unable to get Source", myName))
+	}
+	LogQueryPartMin(esIndex, source, fmt.Sprintf("%s query", myName))
+
+	searchResult, err := client.Search().
+		SearchSource(searchSource).
+		Index(esIndex).
+		Do(context.Background())
+
+	if err != nil {
+		logrus.Error("getProfileSuggestions search failed")
+		return nil, err
+	}
+
+	logrus.Debugf("Search query took %d milliseconds\n", searchResult.TookInMillis)
+
+	if outerControlsAggResult, found := searchResult.Aggregations.Nested("controls"); found {
+		if filteredControls, found := outerControlsAggResult.Aggregations.Filter("filtered_controls"); found {
+			if controlBuckets, found := filteredControls.Aggregations.Terms("control"); found && len(controlBuckets.Buckets) > 0 {
+				for _, controlBucket := range controlBuckets.Buckets {
+					contListItem := reportingapi.ControlItem{}
+					contListItem.Id = controlBucket.Key.(string)
```

Comment:
Should we use the form `contListItem.Id, ok = controlBucket.Key.(string)` here like you do below to ensure this cast doesn't panic.

Reply:
fixing

Reply:
done